File: inject-java/src/test/groovy/io/micronaut/inject/factory/beanwithfactory/DualListener.java
/*
 * Copyright 2017-2020 original authors
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package io.micronaut.inject.factory.beanwithfactory;


import io.micronaut.context.event.BeanCreatedEvent;
import io.micronaut.context.event.BeanCreatedEventListener;
import io.micronaut.context.event.BeanInitializedEventListener;
import io.micronaut.context.event.BeanInitializingEvent;

import javax.inject.Singleton;

@Singleton
public class DualListener implements BeanCreatedEventListener<BFactory>, BeanInitializedEventListener<BFactory> {
    public boolean initialized;
    public boolean created;

    @Override
    public BFactory onCreated(BeanCreatedEvent<BFactory> event) {
        this.created = true;
        return event.getBean();
    }

    @Override
    public BFactory onInitialized(BeanInitializingEvent<BFactory> event) {
        this.initialized = true;
        return event.getBean();
    }
}

File: test-suite-scala/src/test/scala/io/micronaut/inject/factory/beanwithfactory/DualListener.scala
package io.micronaut.inject.factory.beanwithfactory

import io.micronaut.context.event.{BeanCreatedEvent, BeanCreatedEventListener, BeanInitializedEventListener, BeanInitializingEvent}
import javax.inject.Singleton

@Singleton
class DualListener extends BeanCreatedEventListener[BFactory] with BeanInitializedEventListener [BFactory ]{
  var initialized = false
  var created = false

  override def onCreated(event: BeanCreatedEvent[BFactory]): BFactory = {
    this.created = true
    event.getBean
  }

  override def onInitialized(event: BeanInitializingEvent[BFactory]): BFactory = {
    this.initialized = true
    event.getBean
  }
}
